substrate/utils/frame/benchmarking-cli/src/extrinsic/bench.rs

```diff
@@ -142,34 +149,42 @@ where
 			builder.push(inherent)?;
 		}
 
-		// Return early if `ext_builder` is `None`.
-		let ext_builder = if let Some(ext_builder) = ext_builder {
-			ext_builder
-		} else {
-			return Ok((builder.build()?.block, None))
+		let num_ext = match ext_builder {
+			Some(ext_builder) => {
+				// Put as many extrinsics into the block as possible and count them.
+				info!("Building block, this takes some time...");
+				let mut num_ext = 0;
+				for nonce in 0..self.max_ext_per_block() {
+					let ext = ext_builder.build(nonce)?;
+					match builder.push(ext.clone()) {
+						Ok(()) => {},
+						Err(ApplyExtrinsicFailed(Validity(TransactionValidityError::Invalid(
+							InvalidTransaction::ExhaustsResources,
+						)))) => break, // Block is full
+						Err(e) => return Err(Error::Client(e)),
+					}
+					num_ext += 1;
+				}
+				if num_ext == 0 {
+					return Err("A Block must hold at least one extrinsic".into())
+				}
+				info!("Extrinsics per block: {}", num_ext);
+				Some(num_ext)
+			},
+			None => None,
 		};
 
-		// Put as many extrinsics into the block as possible and count them.
-		info!("Building block, this takes some time...");
-		let mut num_ext = 0;
-		for nonce in 0..self.max_ext_per_block() {
-			let ext = ext_builder.build(nonce)?;
-			match builder.push(ext.clone()) {
-				Ok(()) => {},
-				Err(ApplyExtrinsicFailed(Validity(TransactionValidityError::Invalid(
-					InvalidTransaction::ExhaustsResources,
-				)))) => break, // Block is full
-				Err(e) => return Err(Error::Client(e)),
-			}
-			num_ext += 1;
-		}
-		if num_ext == 0 {
-			return Err("A Block must hold at least one extrinsic".into())
-		}
-		info!("Extrinsics per block: {}", num_ext);
+		let proof_size = builder.current_proof_size();
 		let block = builder.build()?.block;
```

Comment:
why don't we include the proof_size for this operation too?

Reply:
This is a good catch, changed!